genesyscloud/task_management_workitem_schema/resource_genesyscloud_task_management_workitem_schema.go

```diff
@@ -0,0 +1,168 @@
+package task_management_workitem_schema
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v115/platformclientv2"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+/*
+The resource_genesyscloud_task_management_workitem_schema.go contains all of the methods that perform the core logic for a resource.
+*/
+
+// getAllTaskManagementWorkitemSchemas retrieves all of the task management workitem schemas via Terraform in the Genesys Cloud and is used for the exporter
+func getAllTaskManagementWorkitemSchemas(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := getTaskManagementProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	schemas, err := proxy.getAllTaskManagementWorkitemSchema(ctx)
+	if err != nil {
+		return nil, diag.Errorf("failed to get all workitem schemas: %v", err)
+	}
+
+	for _, schema := range *schemas {
+		log.Printf("Dealing with task management workitem schema id: %s", *schema.Id)
+		resources[*schema.Id] = &resourceExporter.ResourceMeta{Name: *schema.Id}
+	}
+
+	return resources, nil
+}
+
+// createTaskManagementWorkitemSchema is used by the task_management_workitem_schema resource to create Genesys cloud task management workitem schemas
+func createTaskManagementWorkitemSchema(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTaskManagementProxy(sdkConfig)
+
+	dataSchema, err := BuildSdkWorkitemSchema(d, nil)
+	if err != nil {
+		return diag.Errorf("create: failed to build task management workitem schema: %s", err)
+	}
+
+	log.Printf("Creating task management workitem schema")
+	schema, err := proxy.createTaskManagementWorkitemSchema(ctx, dataSchema)
+	if err != nil {
+		return diag.Errorf("failed to create task management workitem schema: %s", err)
+	}
+
+	d.SetId(*schema.Id)
+
+	// If enabled is set to 'false' do an update call to the schema
+	if enabled, ok := d.Get("enabled").(bool); ok && !enabled {
+		dataSchema.Version = platformclientv2.Int(1)
+		_, err := proxy.updateTaskManagementWorkitemSchema(ctx, *schema.Id, dataSchema)
```

Comment:
Some log statements could be useful here to help track what's going on. Something like "updating task management [...] because xyz" , and then "Updated task management..."

Reply:
Added logging on conditional update